### ISSUE TITLE
ClientModel: Review ClientRequestException constructors per transparency

### ISSUE THREAD

**Issue by annelo-msft** [Azure/azure-sdk-for-net#40666]
As discussed with @KrzysztofCwalina, the constructors we have on ClientRequestException today may not be sufficiently transparent for users to understand the intent of each, how they should be used, and what will happen when called in different ways.  This issue tracks reviewing this and finding a better design for leading users to path of success.

**Comment by annelo-msft** [comment]
Since we filed this, I've updated the ClientResultException constructors such that:
1. `public ClientResultException(PipelineResponse response, Exception? innerException = null)`
  - Creates ClientResultException with `Message` property created from `response`.  Populates the value that will be returned from `GetRawResponse` with `response`.
2. `public ClientResultException(string message, PipelineResponse? response = null, Exception? innerException = null)`
  - Creates ClientResultException with `Message` property created from `message`.  Populates the value that will be returned from `GetRawResponse` with `response` -- which could be `null` if no value is provided.
